src/app.js

```diff
@@ -0,0 +1,247 @@
+            args: ["{that}.tray", "{that}.options.labels.tooltip", "{arguments}.0"] // menu
+        }
+    },
+    icon: "icons/gpii.ico",
+    labels: {
+        tooltip: "GPII Electron"
+    }
+});
+
+gpii.taskTray.makeTray = function (icon) {
+    return new Tray(path.join(__dirname, icon));
+};
+
+gpii.taskTray.updateTaskTray = function (tray, tooltipLabel, menu) {
+    tray.setToolTip(tooltipLabel);
+    tray.setContextMenu(Menu.buildFromTemplate(menu));
+};
+
+/*
+ ** Component to start and stop parts of the GPII as well as key in and key out users
+ */
+fluid.defaults("gpii.app", {
+    gradeNames: "fluid.modelComponent",
+    model: {
+        gpiiStarted: false,
+        keyedInSet: null
+    },
+    events: {
+        onAppReady: null
+    },
+    listeners: {
+        "onCreate.startGpii": {
+            funcName: "gpii.app.startLocalFlowManager",
+            args: ["{that}"]
+        },
+        "onCreate.changeStarted": {
+            funcName: "{that}.changeStarted",
+            args: [true]
+        }
+    },
+    modelListeners: {
+        keyedInSet: [{
+            funcName: "gpii.app.keyOut",
+            args: ["{change}.oldValue"]
+        }, {
+            funcName: "gpii.app.keyIn",
+            args: ["{change}.value"],
+            priority: "after:keyOut"
+        }]
+    },
+    invokers: {
+        changeStarted: {
+            changePath: "gpiiStarted",
+            value: "{arguments}.0"
+        }
+    }
+});
+
+gpii.app.startLocalFlowManager = function (that) {
+    var fluid = require("universal"),
+        gpii = fluid.registerNamespace("gpii");
+
+    if (os.platform() === "win32") {
+        require("gpii-windows/index.js");
+    }
+
+    gpii.start();
+    //TODO: needs to fire an on ready so we can be certain that everything has started
+    that.events.onAppReady.fire();
+};
+
+gpii.app.keyIn = function (token) {
+    request("http://localhost:8081/user/" + token + "/login", function (/*error, response, body*/) {
+        //TODO Put in some error logging
+    });
+};
+
+gpii.app.keyOut = function (token) {
+    request("http://localhost:8081/user/" + token + "/logout", function (/*error, response, body*/) {
+        //TODO Put in some error logging
+    });
+};
+
+/*
+ ** Component to create and update the task tray menu.
+ */
+fluid.defaults("gpii.taskTray.menu", {
+    gradeNames: "fluid.modelComponent",
+    menuLabels: {
+        tooltip: "GPII Electron",
+        keyedIn: "Keyed in with %setName", // string template
+        keyOut: "Key out %setName", //string template
+        notKeyedIn: "Not keyed in",
+        exit: "Exit",
+        keyIn: "Key in set ..."
+    },
+    snapsets: {
+        alice: "Alice",
+        davey: "Davey",
+        david: "David",
+        elaine: "Elaine",
+        elmer: "Elmer",
+        elod: "Elod",
+        livia: "Livia"
+    }
+});
+
+gpii.taskTray.updateSet = function (menu, menuLabels, setName, changeSetFn) {
```

Comment:
This has been addressed through refactoring the menu component. cc504d443bd10954a9c946f753d844a4d7556f32